src/core/video/video.service.ts

```diff
@@ -0,0 +1,58 @@
+import type { ResizeOptions } from 'sharp';
+import type { Readable, Stream } from 'stream';
+import ffmpegBinary from '@ffmpeg-installer/ffmpeg';
+import type { FfprobeData } from 'fluent-ffmpeg';
+import ffmpeg from 'fluent-ffmpeg';
+import fs from 'fs';
+import tmp from 'tmp';
+import util from 'util';
+
+const createTempFilename = util.promisify(tmp.tmpName);
+
+type VideoResizeOptions = {
+  codecName: string;
+} & ResizeOptions
+
+export class VideoService {
+  constructor() {
+    ffmpeg.setFfmpegPath(ffmpegBinary.path);
+  }
+
+  public resize(file: Readable, options: VideoResizeOptions): Stream {
+    return ffmpeg()
+      .input(file)
+      .size(`${options.height}x${options.width}`)
+      .format(options.codecName)
+      .pipe();
+  }
+
+  public async metadata(file: Readable): Promise<FfprobeData> {
+    return await new Promise((resolve, reject) => {
+      ffmpeg().input(file).ffprobe((err, data) => {
+        if (err) {
+          reject(err);
+        }
+
+        resolve(data);
```

Comment:
Do we want to serialize the response? Currently it's raw output which looks like this:
```
{
  "streams": [
    {
      "index": 0,
      "codec_name": "h264",
      "codec_long_name": "H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10",
      "profile": "High",
      "codec_type": "video",
      "codec_tag_string": "avc1",
      "codec_tag": "0x31637661",
      "width": 1920,
      "height": 1080,
      "coded_width": 1920,
      "coded_height": 1080,
      "closed_captions": 0,
      "has_b_frames": 2,
      "sample_aspect_ratio": "1:1",
      "display_aspect_ratio": "16:9",
      "pix_fmt": "yuv420p",
      "level": 40,
      "color_range": "unknown",
      "color_space": "unknown",
      "color_transfer": "unknown",
      "color_primaries": "unknown",
      "chroma_location": "left",
      "field_order": "unknown",
      "refs": 1,
      "is_avc": "true",
      "nal_length_size": 4,
      "id": "N/A",
      "r_frame_rate": "30000/1001",
      "avg_frame_rate": "30000/1001",
      "time_base": "1/30000",
      "start_pts": 0,
      "start_time": 0,
      "duration_ts": 100100,
      "duration": 3.336667,
      "bit_rate": 763463,
      "max_bit_rate": "N/A",
      "bits_per_raw_sample": 8,
      "nb_frames": 100,
      "nb_read_frames": "N/A",
      "nb_read_packets": "N/A",
      "tags": {
        "language": "und",
        "handler_name": "Apple Video Media Handler",
        "vendor_id": "[0][0][0][0]"
      },
      "disposition": {
        "default": 1,
        "dub": 0,
        "original": 0,
        "comment": 0,
        "lyrics": 0,
        "karaoke": 0,
        "forced": 0,
        "hearing_impaired": 0,
        "visual_impaired": 0,
        "clean_effects": 0,
        "attached_pic": 0,
        "timed_thumbnails": 0
      }
    }
  ],
  "format": {
    "filename": "pipe:0",
    "nb_streams": 1,
    "nb_programs": 0,
    "format_name": "mov,mp4,m4a,3gp,3g2,mj2",
    "format_long_name": "QuickTime / MOV",
    "start_time": 0,
    "duration": 3.337,
    "size": "N/A",
    "bit_rate": "N/A",
    "probe_score": 100,
    "tags": {
      "major_brand": "isom",
      "minor_version": "512",
      "compatible_brands": "isomiso2avc1mp41",
      "encoder": "Lavf58.45.100"
    }
  },
  "chapters": []
}
```

Reply:
nevermind, I found the issue with details: https://github.com/leocode/image-service/issues/2